components/motor/gpiostepper/gpiostepper.go

```diff
@@ -262,33 +270,38 @@ func (m *gpioStepper) GoFor(ctx context.Context, rpm, revolutions float64, extra
 
 func (m *gpioStepper) goForInternal(ctx context.Context, rpm, revolutions float64) error {
 	if revolutions == 0 {
-		revolutions = 1000000.0
+		// go a large number of revolutions if 0 is passed in, at the desired speed
+		revolutions = 1000000
 	}
-	var d int64 = 1
 
-	if math.Signbit(revolutions) != math.Signbit(rpm) {
-		d = -1
+	if math.Abs(rpm) < 0.1 {
+		m.logger.Debug("motor (%s) speed less than .1 rev_per_min, stopping", m.motorName)
+		return m.Stop(ctx, nil)
 	}
 
-	revolutions = math.Abs(revolutions)
-	rpm = math.Abs(rpm) * float64(d)
+	var d int64 = 1
 
-	if math.Abs(rpm) < 0.1 {
-		return m.Stop(ctx, nil)
+	if math.Signbit(revolutions) != math.Signbit(rpm) {
+		d = -1
 	}
 
 	m.lock.Lock()
 	defer m.lock.Unlock()
 
+	// calculate delay between steps for the thread in the gorootuine that we started in component creation
+	m.stepperDelay = time.Duration(time.Minute) / time.Duration(math.Abs(rpm)*float64(m.stepsPerRotation))
```

Comment:
Nit:
Argh. I think my last comment on this was bad. Math with Duration is funny.

I think `time.Duration(math.Abs(rpm)*float64(m.stepsPerRotation))` shouldn't be a duration as rpm/stepsPerRotation is not a time unit.

Best I (stackoverflow really) can come up with is : `time.Duration(int64(float64(time.Minute) / (math.Abs(rpm) * float64(stepsPerRotation))))`

Reply:
can confirm, hardware testing now doesn't move motor.

Reply:
fixededed